munit-scalacheck/shared/src/main/scala/munit/scalacheck/ScalaCheckSuite.scala

```diff
@@ -0,0 +1,39 @@
+package munit
+package scalacheck
+
+import org.scalacheck.Prop
+import org.scalacheck.{Test => ScalaCheckTest}
+import org.scalacheck.util.Pretty
+import scala.concurrent.Future
+
+abstract class ScalaCheckSuite extends FunSuite {
+
+  def property(
+      options: TestOptions
+  )(body: => Prop)(implicit loc: Location): Unit = {
+    test(options)(body)
+  }
+
+  override def munitValueTransforms: List[ValueTransform] =
+    super.munitValueTransforms ++ List(
+      new ValueTransform("ScalaCheck Prop", {
+        case prop: Prop =>
+          val result = ScalaCheckTest.check(scalaCheckTestParameters, prop)
+          val prettyResult = Pretty.pretty(result, scalaCheckPrettyParameters)
+          if (result.passed) {
+            println(prettyResult)
+            Future.successful(())
+          } else
+            Future.failed {
+              val e = new Exception("\n" + prettyResult)
```

Comment:
I'm throwing a generic `Exception` here, not sure there's a more appropriate way of doing this